Source/JavaScriptCore/yarr/YarrPattern.cpp

```diff
@@ -1332,10 +1331,38 @@ class YarrPatternConstructor {
     void atomNamedBackReference(const String& subpatternName)
     {
         ASSERT(m_pattern.m_namedGroupToParenIndeces.find(subpatternName) != m_pattern.m_namedGroupToParenIndeces.end());
-        auto namedGroupIndeces = m_pattern.m_namedGroupToParenIndeces.get(subpatternName);
+        auto parenIndeces = m_pattern.m_namedGroupToParenIndeces.get(subpatternName);
+
+        if (parenIndeces.size() == 2) {
+            // If this isn't a duplicate group, we need to go through the same analysis as a non-named backreferece to determine if
+            // this backreference appears in the capture itself. A duplicate could be satisfied by a prior capture and therefore doesn't
+            // need this analysis.
+            unsigned subpatternId = parenIndeces.last();
+
+            PatternAlternative* currentAlternative = m_alternative;
+            ASSERT(currentAlternative);
+
+            while ((currentAlternative = currentAlternative->m_parent->m_parent)) {
```

Comment:
We've asserted that `currentAlternative` is non-null, but do we need to assert similarly for `currentAlternative ->m_parent`?

Reply:
No, there should always be a parent for an alternative.  For a top level alternative.  There is the main disjunction.